lib/components/CircuitJsonPreview/CircuitJsonPreview.tsx

```diff
@@ -104,21 +129,44 @@ export const CircuitJsonPreview = ({
   }
 
   useEffect(() => {
-    if (errorMessage) {
+    if (circuitJsonErrors || errorMessage) {
       setActiveTab("errors")
     }
-  }, [errorMessage])
+  }, [circuitJsonErrors, errorMessage])
 
   useEffect(() => {
     if (
       (activeTab === "code" || activeTab === "errors") &&
       circuitJson &&
-      !errorMessage
+      !circuitJsonErrors
     ) {
       setActiveTab(defaultActiveTab ?? "pcb")
     }
   }, [circuitJson])
 
+  const errorsList = useMemo(() => {
+    const combinedErrorList: { errorTitle: string; errorMessage: string }[] = []
+
+    if (errorMessage) {
+      combinedErrorList.push({
+        errorTitle: "Error",
+        errorMessage: errorMessage,
+      })
+    }
+
+    if (circuitJsonErrors) {
+      for (const { error, errorMessage } of circuitJsonErrors) {
+        const errorTitle = error
+          .split("_")
+          .map((word: string) => word.charAt(0).toUpperCase() + word.slice(1))
+          .join(" ")
+        combinedErrorList.push({ errorTitle: errorTitle, errorMessage })
+      }
+    }
+
+    return combinedErrorList.length > 0 ? combinedErrorList : null
+  }, [errorMessage, circuitJsonErrors])
```

Comment:
You dont need to combine errors. The nature of the errorMessage prop is it is an execution error so there isnt circuit JSON if it is provided. It overrides other error messages.